core/chains/evm/txmgr/txmgr.go

```diff
@@ -424,6 +440,18 @@ func signedTxHash(signedTx *gethTypes.Transaction, chainType chains.ChainType) (
 	return hash, nil
 }
 
+func (b *Txm) MakeForwardedTransaction(tx NewTx) (NewTx, error) {
+	fwds, count, err := b.forwardersMgr.ORM.FindForwardersByChain(utils.Big(b.chainID))
+	if err != nil || count == 0 {
+		return tx, errors.Wrap(err, "Failed to retrieve forwarders")
+	}
+
+	tx.EncodedPayload = b.forwardersMgr.MakeForwardedPayload(tx.ToAddress, tx.EncodedPayload)
+	tx.ToAddress = fwds[0].Address
```

Comment:
nit: Once we have a list of forwarder addresses that can be used, should we randomize which one to use? Or always pick the 1st from this list?

Reply:
Another thing was, once we have the forwarder addresses from the onchain contract, we need to cache them.